bloom-housing/bloom
# 300/request approval email

UPDATE: Going to put all the emails in one PR so back to WIP

# Pull Request Template

## Issue Overview

This PR addresses https://github.com/metrotranscom/doorway/issues/300


## Description

This PR creates a combined endpoint that will update the listing to pendingReview and send a corresponding email to all admins indicating that the listing needs approval. 

Two Notes! 1) As I build out the other emails, I'll explore ways to avoid repeating code (ie. updateAndNotify endpoint that takes in listingStatus and determines what email to send). I decided this would be easier done while developing the others rather than predicting patterns. 2) No QA notes for product since all three will go into dev at once and will develop detailed QA for the whole flow.

## How Can This Be Tested/Reviewed?

Pull down locally
1. Reseed
2. Sign in to admin@example.com
3. Change Test: Coliseum to a draft listing
4. Create an admin account for your work email
5. Sign out
6. Go to work email and confirm + create new account
7. Sign out
8. Sign in to "leasing-agent-1@example.com" with usual password
9. Go to Test: Coliseum and click submit
10. The listing should now say pending review
11. Go to work email and see email that aligns with Figma

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
